packages/interactivity/src/vdom.ts

```diff
@@ -18,16 +18,15 @@ const isObject = ( item: unknown ): item is Record< string, unknown > =>
 
 // Regular expression for directive parsing.
 const directiveParser = new RegExp(
-	`^data-${ p }-` + // ${p} must be a prefix string, like 'wp'.
+	`^${ fullPrefix }` + // ${fullPrefix} is the expected prefix string: "data-wp-".
 		// Match alphanumeric characters including hyphen-separated
 		// segments. It excludes underscore intentionally to prevent confusion.
 		// E.g., "custom-directive".
-		'([a-z0-9]+(?:-[a-z0-9]+)*)' +
+		'([a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*)' +
```

Comment:
Can be non-capturing group, the regex is tested now, the groups are unused.
```suggestion
		'(?:[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*)' +
```